### DISCUSSION TITLE
Does Sample Loaning (not in data-sharing) require bounded type?

### DISCUSSION THREAD

**Discussion by zoeeer** [eProsima/Fast-DDS#5087] [Q&A] [answered]
related discusstions: https://github.com/eProsima/Fast-DDS/discussions/3654

I got the idea that sample loaning is not only possible with data-sharing, but also can be applied to other delivery mechanisms.

What are the constraints to use loaning?
- does it require plain type?
  - I see in the [SampleLoanManager](https://github.com/eProsima/Fast-DDS/blob/08f5f72ed08fac4fa910ffa8baf5e41439f71b26/src/cpp/fastdds/subscriber/DataReaderImpl/SampleLoanManager.hpp#L65) code handles differently for plain and non-plain types. does this mean plain type is not required?
- does it require bounded type?

**Comment by JesusPoderoso** [comment, marked as answer]
Hi @zoeeer, 
There is a distinction between using loans in the DataWriter and the DataReader.
It is mandatory for the type to be plain to use loans in the DataWriter.
It is **NOT** mandatory for the type to be plain to use loans in the DataReader. 
If it is not plain, DataReader requests TypeSupport the data and deserializes that payload.

The bounded type is a data-sharing requirement, not related to loans.
It may not be very clear because the related discussion linked was related to the zero-copy mechanism, which is a combination of loans and data-sharing.
Zero-copy's constraint is the type to be plain. And if the type is plain, it is already bounded (it is a conditional statement: If plain then bounded, but it bounded it is not necessary plain).

So as a summary, loans can always be used in the DataReader. Loans can only be used by a DataWriter when the type is plain. If using loans in DataWriter and DataReader, and if the type is plain, and if the delivery mechanism is data-sharing or intra-process, then the communication would be performed without any additional copy (zero-copy).
